### ISSUE TITLE
Test on the NYUv2 and KITTI

### ISSUE THREAD

**Issue by shenglunch** [Wang-xjtu/G2-MonoDepth#4]
Very nice job! 

I want to cite it in my research. So, I test your model on the NYUv2 dataset like NLSPN (depth completion), but the tested results, RMSE=2.913, AbsRel=0.983, are very different from the reported results in the paper. For KITTI,  the tested results are RMSE=15.967, AbsRel=0.675.

The data setting of NYUv2 is as follows:
Sparsity=1% (693 valid points), H=228, W=304, RGB values are normalized to 0-1, Sparse Depth is normalized by d/20.

The data setting of KITTI is as follows:
Sparsity=64 lines, RGB values are normalized to 0-1, Sparse Depth is normalized by d/100.

Env: python=3.8, torch=2.0

I want to know whether my tested results are reliable.

If possible, could you provide the dataloaders for NYUv2 and KITTI, similar to those of NLSPN?

Thx!

**Comment by Wang-xjtu** [comment]
Hi, thanks for your attention!

Here are some testing suggestions.
1. Have you tested the model on the [test dataset](https://drive.google.com/file/d/1rIkCjvSGQd4b-haedEkLkd7pbJM5hiel/view?usp=drive_link) we provided? These examples will help you check the usage of the model.
2. The resolution of RGB and depth should be a multiple of 64, because we use 6 downsampling layers in the network.
3. You should use "Neareast" to downsample the depth map, because linear interpolation will generate wrong values ​​between valid pixels and invalid pixels.

**Comment by shenglunch** [comment]
Yes! I tested the model on your test dataset and got the same results as in your paper.

I'm very interested in its generalization on the public datasets. The depth maps may not be 16-bit, so I read the sparse depth maps and ground truth in meters. Then, the sparse depth map is normalized by d/max_depth. The prediction is normalized by min_value and max_value, and multiplied by max_depth.

I tested the processing on the Ibims and got rmse=23.9, rel=0.78.

How can I correctly evaluate the model using a depth map in meters?

**Comment by Wang-xjtu** [comment]
So, your problem is that you got the correct results of output but didn't know how to evaluate these results in meters?

The raw depth input of network is `d/max_depth`, so the predicted depth map of network is also `d'/max_depth`.
You can calculate RMSE in meters on `d'` by `RMSE(d'/max_depth*max_depth)`. The output prediction should no longer be normalized.

**Comment by shenglunch** [comment]
Yes, but I'm not sure how to evaluate these results in meters. 

I modify the code using your suggestion, where the point is the raw depth input in meters, the same as gt.

Is it right?

```
x1 = torch.cat((rgb, point/max_depth, hole_point), dim=1)
depth = self.block(x1)*max_depth
Rmse = ((depth - gt)**2)).mean().sqrt()
```

By the way, is the operation `_gt/256_` in the `test_utils.py` used to obtain the depth in meters?

**Comment by Wang-xjtu** [comment]
I think your evaluation code (in meters) is correct.
You can check your results by `AbsRel` in the paper, because `AbsRel `is scale-invariant, i.e. `absrel=|s*pred-s*gt|/(s*gt)=|pred-gt|/gt`.

I think our RMSE code cannot be used for your test. To evaluate all models on multiple datasets, we first store the depth prediction in 16 bits, i.e. `pred/max_depth*65535`. After prediction, our test code is to read and evaluate the stored results. `/256` in `test_utils.py` is to scale the value range from `[0, 65535]` to` [0, 256]` for clearer comparison in the paper.

**Comment by shenglunch** [comment]
Thanks for your patience.

I use depth maps in meters to test your model, as follows:

```
x1 = torch.cat((rgb, point/max_depth, hole_point), dim=1)
depth = self.block(x1)*max_depth
RMSE = ((depth - gt)**2)).mean().sqrt()
AbsREL=(|depth-gt|/gt).mean()

```

For NYUv2, the RMSE = 0.132[m] and REL = 0.019. For KITTI, the RMSE = 1.612[m] and REL = 0.024. For VOID1500, the RMSE = 0.087[m] and REL = 0.021. I think these results are very competitive.
